Babylon/groups/azure/groups/acr/commands/ls.py

```diff
@@ -1,24 +1,38 @@
+import subprocess
 import logging
-from click import command, make_pass_decorator
+from click import command, pass_context, Context, option
 from azure.containerregistry import ContainerRegistryClient
-from Babylon.utils.decorators import requires_external_program, require_deployment_key
+from Babylon.utils.decorators import require_deployment_key
 
 logger = logging.getLogger("Babylon")
 
 """Command Tests
 > babylon acr ls
+Should list all images in the registry given in config
+> babylon acr ls -r my_registry
 Should list all images in the given registry
 Can be checked against `az acr repository list --name my_registry --output table`
 """
 
-pass_cr_client = make_pass_decorator(ContainerRegistryClient)
 
 @command()
-@requires_external_program("docker")
-@pass_cr_client
-@require_deployment_key("acr_registry_name", "acr_registry_name")
-def ls(cr_client: ContainerRegistryClient, acr_registry_name: str):
+@pass_context
+@require_deployment_key("acr_src_registry_name", "acr_src_registry_name")
+@option("-r", "--registry")
+def ls(ctx: Context, acr_src_registry_name: str, registry: str):
```

Comment:
Type of registry should be `typing.Optional[str]` and the `@option` needs a `help` value